kite-data/kite-data-mapreduce/src/main/java/org/kitesdk/data/mapreduce/DatasetKeyOutputFormat.java

```diff
@@ -536,6 +536,7 @@ private static String getTaskAttemptDatasetName(TaskAttemptContext taskContext)
     Class<E> type = getType(jobContext);
 
     String outputUri = conf.get(KITE_OUTPUT_URI);
+    DefaultConfiguration.set(conf);
```

Comment:
Should this be done in `OutputCommitter#setupTask` instead of here? This method could be called more than once, but we want to ensure we set up the environment config just one time per task. The `setupTask` methods seems like a good place, unless you have a reason to do it here?